lib/redis/client.rb

```diff
@@ -120,9 +118,10 @@ def connect
             begin
               call [:auth, username, password]
             rescue CommandError => err # Likely on Redis < 6
-              if err.message.match?(/ERR wrong number of arguments for \'auth\' command/)
+              case err.message
+              when /ERR wrong number of arguments for 'auth' command/
```

Comment:
It doesn't really matter much in this code path, but the `match?` was deliberate to avoid allocating a useless `MatchData`.